File: rust/examples/signer_options/seed_phrase.rs
use dotenv::from_filename;
use near_api::{AccountId, NearToken, NetworkConfig, Signer, Tokens};

#[tokio::main]
async fn main() {
    from_filename("../.env").unwrap();
    let seed_phrase_string = std::env::var("SEED_PHRASE").unwrap();
    let account_id_string = std::env::var("ACCOUNT_ID").unwrap();

    // Create a signer from a seed phrase
    let seed_phrase = Signer::from_seed_phrase(&seed_phrase_string, None).unwrap(); // "royal success river ..."
    let signer = Signer::new(seed_phrase).unwrap(); // Create the signer

    let network = NetworkConfig::testnet();

    let account_id: AccountId = account_id_string.parse().unwrap();

    // Test the signer by transferring NEAR
    Tokens::account(account_id.clone()) // example-account.testnet
        .send_to("receiver-account.testnet".parse().unwrap())
        .near(NearToken::from_near(1))
        .with_signer(signer.clone())
        .send_to(&network)
        .await
        .unwrap()
        .assert_success();
}

File: near-api-js/examples/keystore-options/seed-phrase.js
import dotenv from "dotenv";
import { NEAR } from "near-api-js/tokens";
import { parseSeedPhrase } from "near-api-js/seed-phrase";
import { Account, JsonRpcProvider } from "near-api-js";

// Load environment variables
dotenv.config();
const accountId = process.env.ACCOUNT_ID;

// Create a keystore and add the key pair via a seed phrase
const seedPhrase = process.env.SEED_PHRASE; // "royal success river ..."
const { secretKey } = parseSeedPhrase(seedPhrase);

// Create a connection to testnet RPC
const provider = new JsonRpcProvider({
  url: "https://test.rpc.fastnear.com",
});

// Create an account object
const account = new Account(accountId, provider, secretKey);

// Test the signer by transferring NEAR
const sendTokensResult = await account.transfer({
  token: NEAR,
  amount: NEAR.toUnits("0.1"),
  receiverId: "receiver-account.testnet"
});

console.log(sendTokensResult);
